rasterio/_vsiopener.pyx

```diff
@@ -381,6 +400,8 @@ def _opener_registration(urlpath, obj):
             callbacks_struct.tell = <VSIFilesystemPluginTellCallback>pyopener_tell
             callbacks_struct.seek = <VSIFilesystemPluginSeekCallback>pyopener_seek
             callbacks_struct.read = <VSIFilesystemPluginReadCallback>pyopener_read
+            if hasattr(opener, "read_multi_range"):
+                callbacks_struct.read_multi_range = <VSIFilesystemPluginReadMultiRangeCallback>pyopener_read_multi_range
```

Comment:
I might have registered a callback function that would fail without an opener implementation, but I think your approach is better.

Reply:
maybe we can use this approach for all the other callback (and this will avoid having to deal with the abstract wrapper class. I've also added a `test` for the `read_mulit_range` attribute directly in the callback itself to make sure it will work anytime